src/ast/mod.rs

```diff
@@ -2589,6 +2596,91 @@ pub enum Statement {
         remote_connection: Option<ObjectName>,
     },
+    /// CREATE TRIGGER
+    ///
+    /// Examples:
+    ///
+    /// ```sql
+    /// CREATE TRIGGER trigger_name
+    /// BEFORE INSERT ON table_name
+    /// FOR EACH ROW
+    /// EXECUTE FUNCTION trigger_function();
+    /// ```
+    ///
+    /// Postgres: <https://www.postgresql.org/docs/current/sql-createtrigger.html>
+    CreateTrigger {
+        /// The `OR REPLACE` clause is used to re-create the trigger if it already exists.
+        ///
+        /// Example:
+        /// ```sql
+        /// CREATE OR REPLACE TRIGGER trigger_name
+        /// AFTER INSERT ON table_name
+        /// FOR EACH ROW
+        /// EXECUTE FUNCTION trigger_function();
+        /// ```
+        or_replace: bool,
+        /// The name of the trigger to be created.
+        name: ObjectName,
+        /// Determines whether the function is called before, after, or instead of the event.
+        ///
+        /// Example of BEFORE:
+        ///
+        /// ```sql
+        /// CREATE TRIGGER trigger_name
+        /// BEFORE INSERT ON table_name
+        /// FOR EACH ROW
+        /// EXECUTE FUNCTION trigger_function();
+        /// ```
+        ///
+        /// Example of AFTER:
+        ///
+        /// ```sql
+        /// CREATE TRIGGER trigger_name
+        /// AFTER INSERT ON table_name
+        /// FOR EACH ROW
+        /// EXECUTE FUNCTION trigger_function();
+        /// ```
+        ///
+        /// Example of INSTEAD OF:
+        ///
+        /// ```sql
+        /// CREATE TRIGGER trigger_name
+        /// INSTEAD OF INSERT ON table_name
+        /// FOR EACH ROW
+        /// EXECUTE FUNCTION trigger_function();
+        /// ```
+        period: TriggerPeriod,
+        /// Multiple events can be specified using OR, such as `INSERT`, `UPDATE`, `DELETE`, or `TRUNCATE`.
+        events: Vec<TriggerEvent>,
+        /// The table on which the trigger is to be created.
+        table_name: ObjectName,
+        /// This keyword immediately precedes the declaration of one or two relation names that provide access to the transition relations of the triggering statement.
+        referencing: Vec<TriggerReferencing>,
+        /// This specifies whether the trigger function should be fired once for
+        /// every row affected by the trigger event, or just once per SQL statement.
+        trigger_object: TriggerObject,
+        /// Whether to include the `EACH` term of the `FOR EACH`, as it is optional syntax.
+        include_each: bool,
+        ///  Triggering conditions
+        condition: Option<Expr>,
+        /// Execute logic block
+        exec_body: TriggerExecBody,
+        /// The characteristic of the trigger, which include whether the trigger is `DEFERRABLE`, `INITIALLY DEFERRED`, or `INITIALLY IMMEDIATE`,
+        characteristics: Option<DeferrableCharacteristics>,
```

Comment:
Ah I see, would it make sense to use `ConstraintCharacteristics` directly instead?
 that would also let us use the parser `parse_constraint_characteristics` for it likely, which would be simpler. It would technically contain a superset of what's allowed for triggers in terms of representation but I think that would be okay

Reply:
`ConstraintCharacteristics` has the attribute `enforced`, which would not make sense for a `TRIGGER`. For this reason, I have separated the two structs so that the code redundancy is minimal while not having incorrect attributes in the `CreateTrigger` struct. Using `parse_constraint_characteristics` would allow for parsing of incorrect `CREATE TRIGGER` statements that include `ENFORCED` statements, and would require us to check that such a term is not included, making the overall code more complex.

Reply:
hmm, the issues I saw were that:
- with the changes to `ConstraintCharacteristics`, the representation of that struct becomes somewhat haphazard, previously it was a bit clear how the different fields belong together, now the extra nesting of `deferrable` is arbitrary which is confusing (looking at the struct its not at all clear why deferrable is nested within it, and there's no reason for it to be.)
- re checking `ENFORCED` downstream, I think that's covered by this part of the [project's goals](https://github.com/sqlparser-rs/sqlparser-rs?tab=readme-ov-file#syntax-vs-semantics) - in that a downstream project will likely usually need to do some semantic analysis - keep in mind that in this specific scenario, if e.g. another dialect has different requirements/additional characteristics that need to live in the new  deferrable struct, you'd still end up having to  check in your project after all.

The changes feels to me like they've pushed the complexity into this library rather than handle outside (which would likely be simpler to) - with the changes, we seem to have introduced an arbitrary layer of abstraction/indirection, and introduced duplicate [parsing logic](https://github.com/sqlparser-rs/sqlparser-rs/blob/c043b583c091dc5d1b8bf0d4cff53cc399ae20d5/src/parser/mod.rs#L4208-L4226) for that layer

Reply:
Restored to `ConstraintCharacteristics` in commit 09581e74360d56eec40b7b34cf1767e35b1af702